### PR TITLE
PADV-1475 test: Creating test for shared componets and helper functions.

### PR DESCRIPTION
**Pull request by nandodev-net**
## Tickets

- https://agile-jira.pearson.com/browse/PADV-1475

## Description
This PR aims to improve and finalize the test coverage for the core application components and helper functions. It includes comprehensive test suites for the following components:

1. **Core Components:**

- Main.jsx: Ensures that the main application routing and component loading logic are correctly implemented, covering all possible routes and their respective components (ClassRoster, LabSummary, LabDetails).

2. **Main Views:**

**ClassRoster:**
- Index.jsx: Tests for correct rendering, data fetching, error handling, and interactions.
- Columns.jsx: Tests column definitions and specific cell rendering logic.

**LabSummary:**
- Index.jsx: Includes tests for component rendering, pagination, error handling, and data fetching.
- Columns.jsx: Verifies the column configurations, including custom cell logic for different data types.

**LabDetails:**
- Index.jsx: Ensures that lab details are fetched correctly, errors are managed properly, and navigation actions work as expected.

3. **Helper Functions:**

Helpers/Index.js: Covers all utility functions and ensures their expected behavior across different scenarios.

## Changes Made

This PR targets increasing the test coverage for the critical parts of the application, particularly focusing on routing, main view functionality, and helper functions. Expected coverage:

- Main.jsx: 89.47%
- ClassRoster: Index and Columns at 94.44%
- LabSummary: Index and Columns at 93.18%
- LabDetails: Index at 95.83%
- Helper functions: 100%

## How To Test

1. Ensure your development environment is set up with the latest dependencies by running `npm i` in your MFE folder.
2. Run all test with `npm test -- --verbose`
Verify that all tests pass and that coverage reports indicate the coverage for the targeted files.

### Component Tests

- **Helpers**
    - [x] Index ----> **Done** cov. 100/100

- **Views**
    - ClassRoster
        - [x] Index ----> **Done** cov. 92.85/100
        - [x] Columns ----> **Done** cov. 100/100

    - LabSummary
        - [x] Index ----> **Done** cov. 90.32/100
        - [x] Columns ----> **Done** cov. 100/100

    - LabDetails
        - [x] Index ----> Done cov. 95.83/100
 
- **Core**
    - [x] Main ----> Done cov. 89.47/100
    
    
    
image

### CODE REVIEW

File: src/views/LabSummary/tests/index.test.jsx [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+  it('returns "N/A" when exam_score or exam_max_possible_score is invalid', () => {
+    const laboratory = {
+      exam_passed: false,
+      exam_score: undefined,
+      exam_max_possible_score: undefined,
+    };
+
+    const examPercentage = (laboratory.exam_score / laboratory.exam_max_possible_score) * 10;
+    expect(Number.isNaN(examPercentage)).toBe(true); // Cambia según el comportamiento esperado
```
Comment (01001110J) [inline src/views/LabSummary/tests/index.test.jsx:278]: image

File: src/views/LabSummary/tests/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+      exam_score: laboratory.exam_score ?? 'N/A',
+      exam_percentage: Number.isNaN(examPercentage) ? 'N/A' : `${examPercentage}%` ?? 'N/A',
+      exam_passed: passedValue,
+      start_time: formatUnixTimestamp(laboratory.start_time),
+      end_time: formatUnixTimestamp(laboratory.end_time),
+    });
+
+    // Verifica que se calculen los valores correctos
+    expect(passedValue).toBe('Yes');
+    expect(examPercentage).toBe(8); // Asegúrate de que el valor de porcentaje sea el esperado
```
Comment (01001110J) [inline src/views/LabSummary/tests/index.test.jsx:331]: image

**Comment by JuanDavidBuitrago** [comment]
@nandodev-net You have a lot of unnecessary inline comments, even in Spanish.

**Comment by nandodev-net** [comment]
> @nandodev-net You have a lot of unnecessary inline comments, even in Spanish.

@JuanDavidBuitrago Sure, but given so many tests of that tool that I don't know well, for me they were very useful. Now that I know the approach makes sense, I clean them

File: src/views/LabDetails/index.jsx [hunk 3]
```diff
@@ -9,8 +9,8 @@ import AlertMessage from 'shared/AlertMessage';
 import LabDetailsCard from 'shared/LabDetailsCard';
 import LabDetailsChartCard from 'shared/LabDetailsChartCard';
 import JsonViewer from 'shared/JsonViewer';
-import { formatTime } from 'helpers';
 import { skillableUrl, mfeBaseUrl } from 'constants';
+import { formatTime } from '../../helpers';
```
[the remark this answers is not in the document]
Comment (nandodev-net) [reply src/views/LabDetails/index.jsx:13]: for some reason, it does not work with helpers script

File: src/core/tests/Main.test.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import Main from '../Main';
+import '@testing-library/jest-dom';
+
+// Mock the child components
+jest.mock('../../views/ClassRoster', () => () => <div>Mocked ClassRoster Component</div>);
+jest.mock('../../views/LabSummary', () => () => <div>Mocked LabSummary Component</div>);
+jest.mock('../../views/LabDetails', () => () => <div>Mocked LabDetails Component</div>);
+
+// Mock constants
+jest.mock('constants', () => ({
```
[the remark this answers is not in the document]
Comment (nandodev-net) [reply src/core/tests/Main.test.jsx:12]: I am not interested in involving a component that have his own tests, a mock is enough to demonstrate that has been called.

File: src/views/LabSummary/tests/index.test.jsx [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+  waitFor,
+  act,
+  fireEvent,
+} from '@testing-library/react';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { logError } from '@edx/frontend-platform/logging';
+import LabSummary from '../index';
+import '@testing-library/jest-dom';
+import { columns } from '../columns';
+import { formatUnixTimestamp } from '../../../helpers';
```
[the remark this answers is not in the document]
Comment (01001110J) [reply src/views/LabSummary/tests/index.test.jsx:16]: @nandodev-net

File: src/views/LabDetails/tests/index.test.jsx [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+} from '@testing-library/react';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { logError } from '@edx/frontend-platform/logging';
+import {
+  BrowserRouter as Router,
+  MemoryRouter,
+} from 'react-router-dom';
+import LabDetails from '../index';
+import '@testing-library/jest-dom';
+
```
[the remark this answers is not in the document]
Comment (01001110J) [reply src/views/LabDetails/tests/index.test.jsx]: @nandodev-net

File: jest.config.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
+    // Your existing Jest configuration options
+    moduleNameMapper: {
+      '^views/(.*)$': '<rootDir>/src/views/$1',
+      '^shared/(.*)$': '<rootDir>/src/shared/$1',
+      '^constants/(.*)$': '<rootDir>/src/constants/$1',
+      // Add more mappings as needed
+      '\\.(css|less|scss|sass)$': 'identity-obj-proxy',  // Mock CSS modules
+      '\\.(svg|png|jpg|jpeg|gif)$': '<rootDir>/__mocks__/fileMock.js', // Mock files like images
```
Comment (JuanDavidBuitrago) [inline jest.config.js:9]: These inline comments are necessaries? @nandodev-net

**Review by JuanDavidBuitrago — APPROVED** [review]

**Review by 01001110J — APPROVED** [review]

**Pull request merged**